conda_build/index.py

```diff
@@ -226,1195 +200,46 @@ def get_build_index(subdir, bldpkgs_dir, output_folder=None, clear_cache=False,
 
 
 def _ensure_valid_channel(local_folder, subdir):
-    for folder in {subdir, 'noarch'}:
+    for folder in {subdir, "noarch"}:
         path = os.path.join(local_folder, folder)
         if not os.path.isdir(path):
             os.makedirs(path)
 
 
-def update_index(dir_path, check_md5=False, channel_name=None, patch_generator=None, threads=MAX_THREADS_DEFAULT,
-                 verbose=False, progress=False, hotfix_source_repo=None, subdirs=None, warn=True,
-                 current_index_versions=None, debug=False, index_file=None):
-    """
-    If dir_path contains a directory named 'noarch', the path tree therein is treated
-    as though it's a full channel, with a level of subdirs, each subdir having an update
-    to repodata.json.  The full channel will also have a channeldata.json file.
-
-    If dir_path does not contain a directory named 'noarch', but instead contains at least
-    one '*.tar.bz2' file, the directory is assumed to be a standard subdir, and only repodata.json
-    information will be updated.
-
-    """
-    base_path, dirname = os.path.split(dir_path)
+def update_index(
+    dir_path,
+    check_md5=False,
+    channel_name=None,
+    patch_generator=None,
+    threads=1,
+    verbose=False,
+    progress=False,
+    hotfix_source_repo=None,
+    subdirs=None,
+    warn=True,
+    current_index_versions=None,
+    debug=False,
+    index_file=None,
+):
+    # conda-build calls update_index on a single subdir internally, but
+    # conda-index expects to index every subdir under dir_path
```

Comment:
this is possible used as an api by conda_build's index cli (although I would rather you use conda-index's new CLI with added features)